### ISSUE TITLE
Lets Encrypt /dehydrated not renewing expired certificates due to IPv6 DNS entries - for the third time

### ISSUE THREAD

**Issue by EdNett** [omega8cc/boa#1179]
We have our domains using both IPv4 A records and IPv6 AAAA records at the registrar. Does this cause a problem with Lets Encrypt/dehydrated ? It appears so.  SO, do we have to manually remove all IPv6 DNS entries at the registrar, or can you modify the script adding the parameter to only use IPv4 (or to give IPv4 addresses priority over IPv6 addresses - that is to use IPv4 by default) when issuing or renewing SSL certs?

[hosting_le] The LE Certificate attempted creation failed in /data/disk/o1/tools/le/certs/example.com | -
-- | --
[hosting_le] Make sure that all aliases have valid DNS names pointing to your instance IP address. | -
[hosting_le] The alternative is to disable SAN mode for this site with empty ctrl file: /data/disk/o1/static/control/ssl-no-san-example.com.info

I saw, (but was too slow to copy) a return on the terminal that indicated that let's encrypt saw both the IPv4 and the IPv6 addresses, and was using the IPv6 address for this site.  

This is an information request. Thank you

Ed

#1228 - cannot seem to reopen this issue; perhaps since we didn't close it.

**Comment by EdNett** [comment]
I have another issue about LE for sites that were part of a failed migration from 3.2.0 sable to 3.2.0 stable - we cannot enable ssl with LE successfully - and receive the same error for all sites that were not fully transferred to the target server when the ssh connection was dropped (some sites were fully transferred and the problem does not exist for them):

[hosting_le] Could not symlink cert directory to /data/disk/o1/config/ssl.d/example.com

How can I correct this? I have already tried disabling encryption, waiting a day, etc. and then re-enabling - it's always the same problem with the same error

**Comment by omega8cc** [comment]
We don't support IPv6 yet, and it is certainly not a source of the problem here. Please check the task log and make sure to expand the lines starting with `[hosting_le]` to review detailed report on what exactly failed. You could also try to disable Encryption and re-enable it again on the affected sites.

**Comment by EdNett** [comment]
We're reopening this, because it is still happening on two servers.  
We are using at the registry A entries with a valid ipv4 for 
example.com and www.example.com

AND AAAA entries with a valid ipv6 for
example.com and www.example.com  - is this correct? (that is A AND AAAA entries for each domain)?

the gist of the error is here:
https://gist.github.com/EdNett/dc7f27508156470a325bef3502101f2c

disabling encryption and re-enabling it does not help. Neither does following the advice and creating an empty control file at /data/disk/o1/static/control/ssl-no-san-example.com.info  - we still have the certificate not being accepted when attempting to visit the site over https


Do we need to remove the ipv6  DNS entries at the registrar?  Many domains with both ipv4 and ipv6 entries do work with LE just fine, so I doubt that this is the problem.

**Comment by EdNett** [comment]
Furthermore, I note that there is no example.com.bak entry in
/data/disk/o1/config/ssl.d
for the domain whose LE ssl cert is expired and cannot be renewed (even though there is the example.com entry). All other domains whose LE SSL cert are working have both entries.

**Comment by EdNett** [comment]
We have found that by deleting the IPv6 DNS entries for the domains whose certs are not being renewed, waiting a day (or clearing the DNS cache), then reverifying, allows the certs to be successfully renewed.  WOuldn't it be better to let us keep IPv6 DNS entries at the registrar for BOA domains, and simply add the flag or parameter to the proper script to force AMCE (dehydrated) to only use IPv4 when issuing and renewing certs?  I can see problems with only having IPv4 DNS entries for domains, now and increasingly into the future.  So, I am not closing this issue yet, but will leave it up to you to close or not.

Thanks,

Ed

**Comment by omega8cc** [comment]
Aegir doesn't support IPv6 on the Nginx level yet, so you can't use it anyway, and there is nothing we could fix on the LE script level, since DNS is checked and verified on the LE servers side, not on BOA system side. It seems that the only workaround is to remove IPv6 entries from DNS.

**Comment by omega8cc** [comment]
Hmm.. actually, dehydrated seems to have the options we need:

```
 --ipv4 (-4)                      Resolve names to IPv4 addresses only
 --ipv6 (-6)                      Resolve names to IPv6 addresses only
```

**Comment by EdNett** [comment]
This problem has re-occurred, and it's a real problem.  We believe that we need to have both IPv4 and IPv6 addresses for our boa hosted domains set at the registrar's DNS, because this is becoming necessary more and more, as more and more internet related things use IPv6. You must have made a change once again, and forgot that you need to ensure that only IPv4 addresses are used when renewing a LE certificate. Can you please hot-fix this. We are using recent 3.2.2 stable with php 7.0 and mariadb 10.2 only. Thank you.

**Comment by EdNett** [comment]
HOW do I reopen this issue, since it has happened again - for the third time?

**Comment by omega8cc** [comment]
The fix has been added to use `--ipv4` flag and it should work. Perhaps we should update the `dehydrated` script in use, which we have tried to avoid due to changes breaking backward compatibility.

**Comment by EdNett** [comment]
Hi,

This problem is fixed for us using the 3.2.2 stable BOA. Perhaps when you decide to move to ACME v2, we can apply the fix in the new/updated dehydrated script. We note that another issue (IDN support) has also been fixed in dehydrated and LE. Thanks. I think the important thing is to remember to make sure the --ipv4 flag exists and is being used each time you modify the dehydrated script.

**Comment by omega8cc** [comment]
There was a bug which prevented --ipv4 flag from working (a missing space in the front, ouch!) but it has been fixed already.
